### PR TITLE
PTR20

### PR DESCRIPTION
**Pull request by artem-dehnych**

### CODE REVIEW

File: src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java [hunk 1]
```diff
@@ -60,9 +58,30 @@ private BeanDefinition createBeanDefinition(Class<?> aClass) {
...
                 .withIsPrimary(isPrimary(aClass))
+                .withDependsOn(getDependsOn(aClass))
                 .createBeanDefinitionImpl();
     }
 
+    private Class<?>[] getDependsOn(Class<?> aClass) {
+        var constructors = aClass.getDeclaredConstructors();
+        var constrCount = constructors.length;
+        if (constrCount > 1) {
+            throw new ClassConctructorException("Class have more than one constructor");
```
Comment (mitssev) [inline src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java]: Id suggest to add some specific information, like class name or so

File: src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java [hunk 2]
```diff
@@ -60,9 +58,30 @@ private BeanDefinition createBeanDefinition(Class<?> aClass) {
...
+        } else if (constrCount==0) {
+            throw new ClassConctructorException("Class have not public constructor");
```
Comment (mitssev) [inline src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java]: same here

**Review by mitssev — DISMISSED** [review]

File: src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java (hunk 1 above)
Comment (artem-dehnych) [reply src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java]: fixed

File: src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java (hunk 2 above)
Comment (artem-dehnych) [reply src/main/java/com/petros/bring/reader/impl/AnnotatedBeanDefinitionReader.java]: fixed

**Review by mykolad4 — APPROVED** [review]

**Review by mitssev — APPROVED** [review]

**Pull request merged**